src/core/Utils.js

```diff
@@ -353,6 +354,30 @@ function findDOMElement (element) {
   }
 }
 
+function getSocketHost (url) {
+  // get the host domain
+  var regex = /^(?:https?:\/\/|\/\/)?(?:[^@\n]+@)?(?:www\.)?([^\n]+)/
+  var host = regex.exec(url)[1]
+  var socketProtocol = location.protocol === 'https:' ? 'wss' : 'ws'
+
+  return `${socketProtocol}://${host}`
+}
+
+function _emitSocketProgress (uppyInstance, progressData, file) {
```

Comment:
Can you rename `uppyInstance` to something like `uploader`? I think that's a bit clearer, I thought it was referring to a `Core` instance first.